src/operator/elemwise_op_common.h

```diff
@@ -200,21 +213,49 @@ struct ElemwiseGradUseInOut {
 struct ElemwiseGradUseNone {
   const char *op_name;
   std::vector<nnvm::NodeEntry> operator()(const nnvm::NodePtr& n,
-                                          const std::vector<nnvm::NodeEntry>& ograds) {
+                                          const std::vector<nnvm::NodeEntry>& ograds) const {
     return MakeNonlossGradNode(op_name, n, ograds, {}, n->attrs.dict);
   }
 };
 
 struct CloneGradient {
   const char *op_name;
   std::vector<nnvm::NodeEntry> operator()(const nnvm::NodePtr& n,
-                                          const std::vector<nnvm::NodeEntry>& ograds) {
+                                          const std::vector<nnvm::NodeEntry>& ograds) const {
     std::vector<nnvm::NodeEntry> ret;
-    for (size_t i = 0; i < n->inputs.size(); ++i)
+    const size_t input_count = n->inputs.size();
+    ret.reserve(input_count);
+    for (size_t i = 0; i < input_count; ++i)
       ret.emplace_back(ograds[0]);
     return ret;
   }
 };
+
+/*! \brief Generic conversion of F<OP> kernel mapping to Kernel::Launch mapping */
+template<typename OP, int req>
+struct OpWithReq {
```

Comment:
move these two struct to mxnet_op.h and conform with naming standard there

Reply:
Can you suggest a name? It's not clear how that naming convention maps to this.

Reply:
Can you suggest a name? It's not clear how that naming convention maps to this.

Reply:
It's really not clear what you mean here -- there's only fill and set_zero in mxnet_op.  Not much precedence to go on there.  I rename to op_with_req and mshadow_to_kernel.